src/Infrastructure/Field/tests/ESMF_FieldIOUTest.F90

```diff
@@ -516,9 +551,9 @@ program ESMF_FieldIOUTest
            overwrite=.true., rc=rc)
     end select
 #if (defined ESMF_PIO && ( defined ESMF_NETCDF || defined ESMF_PNETCDF))
-    if(rc.ne.ESMF_SUCCESS) then
-      countfail = countfail + 1
-    endif
+    write(name, *) "Write field_time.nc at time t=",t," in a loop"
+    write(failMsg, *) "Did not return ESMF_SUCCESS"
+    call ESMF_Test((rc==ESMF_SUCCESS), name, failMsg, result, ESMF_SRCLINE)
```

Comment:
The reason for the "countfail" approach is that there is a script to count the number of tests by reading the "!NEX_UTest_Multi_Proc_Only" directives in the source code.  I think this will over-report PASSES.  Basically, we need one call to ESMF_Test(...) for each "!NEX_UTest_Multi_Proc_Only".  It is a limitation of our test system that should eventually be addressed.

Reply:
I couldn't figure out which iteration of the loop was a problem until i changed this test.

Reply:
Okay, in any case, we can only have ONE call to ESMF_Test.  So if a loop iteration passes, then it should not call ESMF_Test(true) until the entire loop succeeds.  Otherwise our test stats will be off.